src/python/pants/util/contextutil.py

```diff
@@ -139,11 +131,13 @@ def stdio_as(stdout_fd, stderr_fd, stdin_fd):
 
 
 @contextmanager
-def signal_handler_as(sig, handler):
+def signal_handler_as(
+  sig: int, handler: Union[int, Callable[[int, FrameType], None]]
```

Comment:
FYI I found this through Typeshed: https://github.com/python/typeshed/blob/717cd1f96f1df3f00b9e5e569d40ae5172ffd800/stdlib/3/signal.pyi#L72

Whenever there are things I don't know how to type, particularly things that are from the stdlib, I open up the Typeshed repo in GitHub and search for the filename of that module.